devito/function.py

```diff
@@ -744,6 +751,26 @@ def dt2(self):
 
         return self.diff(_t, _t).as_finite_difference(indt)
 
+    def argument_values(self, alias=None, **kwargs):
+        """
+        Returns a map of argument values after evaluating user input.
+
+        :param kwargs: Dictionary of user-provided argument overrides.
+        :param alias: (Optional) name under which to store values.
+        """
+        saves = {True: 'int', False: 'None'}
+        # Check if data has the right dimension
+        if self.name in kwargs:
+            new = kwargs.get(self.name)
+            if ~isinstance(new, TimeFunction):
```

Comment:
"not" is the Devito way, not "~"

Reply:
Oh sorry, old habit will fix.

Reply:
squash this and the following condition as
``if isinstance(new, TimeFunction) and new.save != self.save:``